src/dotnet/install.sh

```diff
@@ -4,489 +4,133 @@
+    # The quality option is only used when channel is 'A.B' or 'A.B.Cxx' because it can't be used with STS or LTS
+    #
+    # This script aims to reduce these combinations of options to a single 'version' input
+    # Currently this script does not make it possible to request a version in the form 'A.B' or 'A.B.Cxx' and a quality other than 'GA'
+    if [[ "$version" == "latest" ]]; then
+        # Fetch the latest version manually, because dotnet-install.sh does not support it directly
+        version=$(fetch_latest_sdk_version)
+        channel=""
+    elif [[ "$version" == "lts" ]]; then
+        # When user input is 'lts'
+        # Then version=latest, channel=LTS
+        channel="LTS"
+        version="latest"
+    elif [[ "$version" =~ ^[0-9]+\.[0-9]+$ ]]; then
+        # When user input is form 'A.B' like '3.1'
+        # Then version=latest, channel=3.1
+        channel="$version"
+        version="latest"
+    elif [[ "$version" =~ ^[0-9]+\.[0-9]+\.[0-9]xx$ ]]; then
+        # When user input is form 'A.B.Cxx' like '6.0.4xx'
+        # Then version=latest, channel=6.0.4xx
+        channel="$version"
+        version="latest"
     else
-        check_packages libssl3
-    fi
-
-    get_full_version_details "${sdk_or_runtime}" "${version}"
-
-    DOTNET_INSTALL_PATH="${TARGET_DOTNET_ROOT}/${DOTNET_DOWNLOAD_VERSION}"
-    if [ -d "${DOTNET_INSTALL_PATH}" ]; then
-        echo "(!) Dotnet version ${DOTNET_DOWNLOAD_VERSION} already exists."
-        exit 1
-    fi
-    # exports DOTNET_DOWNLOAD_URL, DOTNET_DOWNLOAD_HASH, DOTNET_DOWNLOAD_NAME
-    echo "DOWNLOAD LINK: ${DOTNET_DOWNLOAD_URL}"
-
-    # Setup the access group and add the user to it.
-    umask 0002
-    if ! cat /etc/group | grep -e "^${ACCESS_GROUP}:" > /dev/null 2>&1; then
-        groupadd -r "${ACCESS_GROUP}"
-    fi
-    usermod -a -G "${ACCESS_GROUP}" "${USERNAME}"
-
-    # Download the .NET binaries.
-    echo "DOWNLOADING BINARY..."
-    TMP_DIR="/tmp/dotnetinstall"
-    mkdir -p "${TMP_DIR}"
-    curl -sSL "${DOTNET_DOWNLOAD_URL}" -o "${TMP_DIR}/${DOTNET_DOWNLOAD_NAME}"
-
-    # Get checksum from .NET CLI blob storage using the runtime version and
-    # run validation (sha512sum) of checksum against the expected checksum hash.
-    echo "VERIFY CHECKSUM"
-    cd "${TMP_DIR}"
-    echo "${DOTNET_DOWNLOAD_HASH} *${DOTNET_DOWNLOAD_NAME}" | sha512sum -c -
-
-    # Extract binaries and add to path.
-    mkdir -p "${DOTNET_INSTALL_PATH}"
-    echo "Extract Binary to ${DOTNET_INSTALL_PATH}"
-    tar -xzf "${TMP_DIR}/${DOTNET_DOWNLOAD_NAME}" -C "${DOTNET_INSTALL_PATH}" --strip-components=1
-
-    CURRENT_DIR="${TARGET_DOTNET_ROOT}/current"
-    if [[ ! -d "${CURRENT_DIR}" ]]; then
-        ln -s "${DOTNET_INSTALL_PATH}" "${CURRENT_DIR}" 
+        # Assume version is an exact version string like '6.0.412' or '8.0.100-rc.1.23371.5''
+        channel=""
     fi
 
-    # Give write permissions to the user.
```

Comment:
Won't we still need this?

Reply:
I believe we would still need the code for updating user & installed dir permissions.

Reply:
This question I don't know much about. All I can say is the scenario tests run under the `vscode` remote user and they pass. I think maybe the `dotnet-install.sh` script ensures correct permissions?

Reply:
> I think maybe the dotnet-install.sh script ensures correct permissions?

I am not too confident about that, but doesn't seem like the script does that. 👇 

![image](https://github.com/devcontainers/features/assets/24955913/977607de-8257-4fd2-b043-12adfc9c4181)

In the past, we intentionally added privs - set user permissions, created specific gids/groups for each Feature and added sticky bits as there were permissions issues. I believe, we still need this. // cc @Chuxel @bamurtaugh

Reply:
Do you recall what permission issues? I'm hesitant to potentially weaken security when currently all tests pass. I think `rwx` for root and `r-x` for non-root users is already an ideal situation.

Reply:
Relevant issue regarding permissions: https://github.com/devcontainers/features/issues/293

The bug reveals itself when you run `dotnet restore`. @sliekens Do any of your tests use `dotnet restore`? Seems like no. Could we add a project with a simple dependency to make sure that works? Here's an example from my original PR: https://github.com/devcontainers/features/pull/467/files#diff-5d8e34cd2b389901f42c076a13c3f622cec4aa490c1eb3c07acb32a324343527

Reply:
@jungaretti there is an implicit restore behind `dotnet run` in the tests. Indeed, I borrowed your example project with the Newtonsoft.Json dependency. Executing `dotnet restore` before `dotnet run` is not strictly needed unless there is a specific reason.

Reply:
Rock on, what a legendary PR!

I confirmed that your feature fixes #293. I made a simple project that consumes this feature. You can make a codespace for my branch here: https://github.com/codespaces/new/?skip_quickstart=true&repo=449816349&ref=sliekens-dotnet&devcontainer_path=.devcontainer%2Fdevcontainer.json

Reply:
Thanks @jungaretti for providing and validating the `dotnet restore` scenario. Also, this bit was added so that a non-root user don't require `sudo` and there were issues with `dotnet nuget` command.

Looping in @bderusha who first contributed this .NET Feature (when it was known as scripts) to [microsoft/vscode-dev-containers](https://github.com/microsoft/vscode-dev-containers/blob/main/script-library/dotnet-debian.sh). 

@bderusha would highly appreciate your review on this PR and your thoughts on removing the permissions bit which was added in https://github.com/microsoft/vscode-dev-containers/pull/1210

Reply:
I contributed this feature a while ago, so forgive the fuzzy memory, but I believe we added this to support running the CLI as non-root without `sudo`.

We also copied the majority of the lines from the [Go install script](https://github.com/devcontainers/features/blob/e39d87bb202a1704e5d15377be14804479e2aebc/src/go/install.sh#L138-L144) so it is certainly possible that some of the lines required for that script made it into ours unnecessarily.

Reply:
So what is the verdict? Personally I don't think it's needed anymore. I couldn't find anything that is blocked by permissions, all tests pass with the current permission set. I'm not comfortable putting these lines of code in without a failure scenario.

Reply:
Agreed. I feel confident we don't need these lines, especially with the codespace repro that proves the original problem is solved.